admin-dev/themes/new-theme/js/components/modal.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+const $ = window.$;
+
+/**
+ * ConfirmModal component
+ *
+ * @param {String} id
+ * @param {String} confirmTitle
+ * @param {String} confirmMessage
+ * @param {String} closeButtonLabel
+ * @param {String} confirmButtonLabel
+ * @param {String} confirmButtonClass
+ * @param {Boolean} closable
+ * @param {Function} confirmCallback
+ *
+ */
+export default function ConfirmModal(params, confirmCallback) {
+  // Construct the modal
+  const {id, closable} = params;
+  const modal = new Modal(params);
+  Object.assign(this, modal);
+
+  // jQuery modal object
+  this.$modal = $(this.modal);
+
+  this.show = () => {
+    this.$modal.modal();
+  };
+
+  this.modalConfirmButton.addEventListener('click', confirmCallback);
+
+  this.$modal.modal({
+    backdrop: (closable ? true : 'static'),
+    keyboard: closable !== undefined ? closable : true,
+    closable: closable !== undefined ? closable : true,
+    show: false,
+  });
+
+  this.$modal.on('hidden.bs.modal', () => {
+    document.querySelector(`#${id}`).remove();
+  });
+
+  document.body.appendChild(this.modal);
+}
+
+/**
+ * Modal component to improve lisibility by constructing the modal outside the main function
+ *
+ * @param {String} id
+ * @param {String} confirmTitle
+ * @param {String} confirmMessage
+ * @param {String} closeButtonLabel
+ * @param {String} confirmButtonLabel
+ * @param {String} confirmButtonClass
+ * @param {Boolean} closable
+ * @param {Function} confirmCallback
+ *
+ */
+function Modal({id = 'confirm_modal', confirmTitle, confirmMessage = '', closeButtonLabel = 'Close', confirmButtonLabel = 'Accept', confirmButtonClass = 'btn-primary'}) {
+  // Main modal element
+  this.modal = document.createElement('div');
```

Comment:
no need to use `this` in this function

Reply:
Return an object in this function. something like

```js
return {modal: {
  dialog: ...,
  confirmation: ....,
}
```
don't waste time by repeating  `modal`  in variables names